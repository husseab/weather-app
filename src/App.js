

function App() {
  return (
    <div className="App">
    <h1>Weather App</h1>
    </div>
  );
}

export default App;

// Update App.js so that it does not use or render the SVG we just deleted. 
// Then delete markup in <App />, keeping only the outermost <div/> with className prop, 
// and inside it create a simple <h1> displaying the words "Weather App".
// Remove styles from App.css. We will add custom styles later.