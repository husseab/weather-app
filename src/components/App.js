import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "./getForecast";
import SearchForm from "./SearchForm";
import "../styles/App.css";

function App() {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  const handleCitySearch = () => {
    getForecast(searchText, setSelectedDate, setForecasts, setLocation);
  };
  useEffect(() => {
    getForecast(searchText, setSelectedDate, setForecasts, setLocation);
  }, [searchText]);
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  return (
    <div className="weather-app">
      <LocationDetails city={location.city} country={location.country} />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
};
