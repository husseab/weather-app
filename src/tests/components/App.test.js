import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import App from "../../components/App";
import forecast from "../../data/forecast.json";

describe("App", () => {
  test("renders App component correctly", () => {
    const { asFragment } = render(
      <App
        location={forecast.location}
        forecasts={forecast.forecasts}
        forecast={forecast.forecasts[0]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
