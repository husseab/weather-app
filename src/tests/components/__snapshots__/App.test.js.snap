// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      class="location"
    >
      ,  
    </h1>
    <div
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
